kernel/kernel-api/src/main/java/io/delta/kernel/types/StructField.java

```diff
@@ -140,14 +136,17 @@ public List<TypeChange> getTypeChanges() {
   }
 
   public boolean isMetadataColumn() {
-    return metadata.contains(IS_METADATA_COLUMN_KEY)
-        && (boolean) metadata.get(IS_METADATA_COLUMN_KEY);
+    return metadata.contains(METADATA_TYPE_KEY);
   }
 
   public boolean isDataColumn() {
     return !isMetadataColumn();
   }
 
+  public MetadataColumnType getMetadataColumnType() {
+    return metadata.getMetadataType(METADATA_TYPE_KEY);
+  }
```

Comment:
I decided against returning an `Optional` here to stay consistent with other methods in this class, and this `isMetadataColumn()` allows you to verify that this won't be `null`.